File: LeetCode/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) 
    {
        HashMap<Integer, Integer> cache = new HashMap<>();

        for(int i = 0; i < nums.length; ++i)
        {
            int key = target - nums[i];
            if(cache.containsKey(key))
                return new int[] {i, cache.get(key)};
            
            cache.put(nums[i], i);
        } 
        return new int[0];   
    }
}

File: LeetCode/TwoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) 
    {
        Dictionary<int, int> values = new();

        for(int i = 0; i < nums.Length; ++i)
        {
            int key = target - nums[i]; 
            if(values.ContainsKey(key))
                return new int[] {i, values[key]};

            values[nums[i]] = i;
        }
        return new int[0];
    }
}
